datadog/fwprovider/resource_datadog_integration_aws_account.go

```diff
@@ -0,0 +1,902 @@
+	ResourcesConfig *resourcesConfigModel `tfsdk:"resources_config"`
+	TracesConfig    *tracesConfigModel    `tfsdk:"traces_config"`
+}
+
+type authConfigModel struct {
+	AwsAuthConfigKeys *awsAuthConfigKeysModel `tfsdk:"aws_auth_config_keys"`
+	AwsAuthConfigRole *awsAuthConfigRoleModel `tfsdk:"aws_auth_config_role"`
+}
+
+type awsAuthConfigKeysModel struct {
+	AccessKeyId     types.String `tfsdk:"access_key_id"`
+	SecretAccessKey types.String `tfsdk:"secret_access_key"`
+}
+
+type awsAuthConfigRoleModel struct {
+	ExternalId types.String `tfsdk:"external_id"`
+	RoleName   types.String `tfsdk:"role_name"`
+}
+
+type awsRegionsModel struct {
+	IncludeAll  types.Bool `tfsdk:"include_all"`
+	IncludeOnly types.List `tfsdk:"include_only"`
+}
+
+type logsConfigModel struct {
+	LambdaForwarder *lambdaForwarderModel `tfsdk:"lambda_forwarder"`
+}
+
+type lambdaForwarderModel struct {
+	Lambdas types.List `tfsdk:"lambdas"`
+	Sources types.List `tfsdk:"sources"`
+}
+
+type metricsConfigModel struct {
+	AutomuteEnabled         types.Bool             `tfsdk:"automute_enabled"`
+	CollectCloudwatchAlarms types.Bool             `tfsdk:"collect_cloudwatch_alarms"`
+	CollectCustomMetrics    types.Bool             `tfsdk:"collect_custom_metrics"`
+	Enabled                 types.Bool             `tfsdk:"enabled"`
+	TagFilters              []*tagFiltersModel     `tfsdk:"tag_filters"`
+	NamespaceFilters        *namespaceFiltersModel `tfsdk:"namespace_filters"`
+}
+
+type tagFiltersModel struct {
+	Namespace types.String `tfsdk:"namespace"`
+	Tags      types.List   `tfsdk:"tags"`
+}
+
+type namespaceFiltersModel struct {
+	ExcludeOnly types.List `tfsdk:"exclude_only"`
+	IncludeOnly types.List `tfsdk:"include_only"`
+}
+
+type resourcesConfigModel struct {
+	CloudSecurityPostureManagementCollection types.Bool `tfsdk:"cloud_security_posture_management_collection"`
+	ExtendedCollection                       types.Bool `tfsdk:"extended_collection"`
+}
+
+type tracesConfigModel struct {
+	XrayServices *xrayServicesModel `tfsdk:"xray_services"`
+}
+
+type xrayServicesModel struct {
+	IncludeAll  types.Bool `tfsdk:"include_all"`
+	IncludeOnly types.List `tfsdk:"include_only"`
+}
+
+func NewIntegrationAwsAccountResource() resource.Resource {
+	return &integrationAwsAccountResource{}
+}
+
+func (r *integrationAwsAccountResource) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	providerData, _ := request.ProviderData.(*FrameworkProvider)
+	r.Api = providerData.DatadogApiInstances.GetAWSIntegrationApiV2()
+	r.Auth = providerData.Auth
+}
+
+func (r *integrationAwsAccountResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = "integration_aws_account"
+}
+
+func (r *integrationAwsAccountResource) ConfigValidators(ctx context.Context) []resource.ConfigValidator {
+	return []resource.ConfigValidator{
+		resourcevalidator.ExactlyOneOf(
```

Comment:
Some of these `ExactlyOneOf`s seem unnecessary. I don't really understand how this is working though. Why does AWSRegionsPath need one but not AWSMetricsPath? I'd think either both would need one or neither?

Reply:
Discussed offline. They are necessary.